include/boost/histogram/utility/clopper_pearson_interval.hpp

```diff
@@ -0,0 +1,58 @@
+// Copyright 2022 Jay Gohil, Hans Dembinski
+//
+// Distributed under the Boost Software License, version 1.0.
+// (See accompanying file LICENSE_1_0.txt
+// or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_HISTOGRAM_UTILITY_CLOPPER_PEARSON_INTERVAL_HPP
+#define BOOST_HISTOGRAM_UTILITY_CLOPPER_PEARSON_INTERVAL_HPP
+
+#include <boost/histogram/utility/binomial_proportion_interval.hpp>
+#include <cmath>
+#include <utility>
+
+namespace boost {
+namespace histogram {
+namespace utility {
+
+template <class ValueType>
+class clopper_pearson_interval : public binomial_proportion_interval<ValueType> {
+  using base_t = binomial_proportion_interval<ValueType>;
+
+public:
+  using value_type = typename base_t::value_type;
+  using interval_type = typename base_t::interval_type;
+
+  explicit clopper_pearson_interval(deviation d = deviation{1.0}) noexcept
+      : z_{static_cast<double>(d)} {}
+
+  interval_type operator()(value_type successes, value_type failures) const noexcept {
+    const value_type half{0.5}, one{1.0};
+    const value_type ns = successes, nf = failures;
+    const value_type n = ns + nf;
+    // Source: https://en.wikipedia.org/wiki/Binomial_proportion_confidence_interval
+    const value_type alpha1 = (one - d) * half;
+    const value_type alpha2 = one - ((one - d) * half);
+    // Source: https://en.wikipedia.org/wiki/Binomial_proportion_confidence_interval#Clopper%E2%80%93Pearson_interval
+    const value_type m1 = ns;
+    const value_type n1 = n - ns + one;
+    const value_type m2 = ns + 1;
+    const value_type n2 = n - ns;
+    // Source: https://en.wikipedia.org/wiki/Beta_distribution
+    const value_type beta1 = (tgamma(m1)*tgamma(n1)) / tgamma(m1+n1);
+    const value_type beta2 = (tgamma(m2)*tgamma(n2)) / tgamma(m2+n2);
```

Comment:
I am not sure what you are calculating here. You need the quantiles of the beta distribution to compute the interval.

https://en.wikipedia.org/wiki/Beta_distribution

You can get those from Boost Math. Look at the documentation of Boost Math:
https://www.boost.org/doc/libs/1_79_0/libs/math/doc/html/dist.html for general advice and
https://www.boost.org/doc/libs/1_79_0/libs/math/doc/html/math_toolkit/dist_ref/dists/beta_dist.html for the beta distribution.
The docs have many examples.

Reply:
With Boost Math, the implementation should be easy, you just need to call the quantile with the arguments listed here https://en.wikipedia.org/wiki/Binomial_proportion_confidence_interval#Clopper%E2%80%93Pearson_interval

Reply:
It's for calculating B(α, β) for f(x;α,β) for the intervals; = (α+1)!*(β+1)! / (α+β+1)! and tgamma(x) = (x-1)!

(Can explain in a meet)

Reply:
This is not calculating B(alpha, beta). B(alpha, beta) is the quantile of a pdf which does not have a closed form. You cannot implement the formula this way. In fact, you cannot implement the formula in code at all. Please read the references that I linked to.

Reply:
Your only option is to use the beta distribution from Boost Math, so please do that instead of trying other solutions that do not work.